### PR TITLE
CleanUp - Remove static app context access (EXPOSUREAPP-4506)

### PR DESCRIPTION
**Pull request by BMItr**
This PR removes the static app context access and some cleanup regarding FormatterHelper, FormatterSubmissionHelper and related.
Details: EXPOSUREAPP-4506

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/view/TestResultSectionView.kt [hunk 1]
```diff
@@ -26,11 +26,10 @@ constructor(
     attrs: AttributeSet? = null,
     defStyleAttr: Int = 0
 ) : ConstraintLayout(context, attrs, defStyleAttr) {
-    private val binding: ViewTestResultSectionBinding
+    private val binding = ViewTestResultSectionBinding.bind(this)
 
     init {
         inflate(context, R.layout.view_test_result_section, this)
-        binding = ViewTestResultSectionBinding.bind(this)
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/view/TestResultSectionView.kt:33]: was not this a fix? Please have a look  #2039

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/view/TestResultSectionView.kt [hunk 2]
```diff
@@ -86,7 +85,7 @@ constructor(
 
                 DeviceUIState.PAIRED_POSITIVE,
                 DeviceUIState.PAIRED_POSITIVE_TELETAN,
-                DeviceUIState.PAIRED_NEGATIVE -> SpannableString(formatTestResult(uiState))
+                DeviceUIState.PAIRED_NEGATIVE -> formatTestResult(context, uiState)
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/view/TestResultSectionView.kt:88]: Also this was introduced here 
https://github.com/corona-warn-app/cwa-app-android/pull/2039/files#diff-f199e04acbe811786cab2f57926351cb55e3700f4449ef116a2dfa18a2eda979R89

File: Corona-Warn-App/src/main/res/layout/include_main_overview_row.xml [hunk 3]
```diff
@@ -43,7 +43,7 @@
                 android:layout_height="@dimen/icon_size_main_card"
                 android:importantForAccessibility="no"
                 android:src="@{icon}"
-                android:tint="@{FormatterHelper.formatColorIcon(iconTint)}"
+                android:tint="@{(iconTint != 0) ? iconTint : @color/colorAccentTintIcon}"
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/layout/include_main_overview_row.xml:46]: Maybe keep the method `formatColor` and provide the fallback color resource as a parameter. 
and can be used in here and  include_divider.xml

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/view/TestResultSectionView.kt (hunk 1 above)
Comment (chiljamgossow) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/view/TestResultSectionView.kt:33]: Yes, that makes the app crash. The layout must first be inflated.

Comment (BMItter) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/view/TestResultSectionView.kt:33]: good catch. had old base in my stash.
will adjust.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/view/TestResultSectionView.kt (hunk 2 above)
Comment (BMItter) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/view/TestResultSectionView.kt]: adjusted to newest 1.11 base

File: Corona-Warn-App/src/main/res/layout/include_main_overview_row.xml (hunk 3 above)
Comment (BMItter) [reply Corona-Warn-App/src/main/res/layout/include_main_overview_row.xml:46]: method is buggy. better not.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/formatter/FormatterHelper.kt [hunk 4]
```diff
@@ -37,12 +17,11 @@ fun formatDrawable(value: Boolean, drawableTrue: Int, drawableFalse: Int): Drawa
  * @param colorFalse
  * @return
  */
-fun formatColor(value: Boolean, colorTrue: Int, colorFalse: Int): Int {
-    val appContext = CoronaWarnApplication.getAppContext()
+fun formatColor(context: Context, value: Boolean, colorTrue: Int, colorFalse: Int): Int {
```
Comment (chiljamgossow) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/formatter/FormatterHelper.kt:20]: Could make it an extention function, e.g. Context.formatColor

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/formatter/FormatterInformationLegalHelper.kt [hunk 5]
```diff
@@ -15,9 +15,10 @@ import de.rki.coronawarnapp.util.device.DefaultSystemInfoProvider
  */
 
 fun formatVisibilityLanguageBased(
+    context: Context,
```
Comment (chiljamgossow) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/formatter/FormatterInformationLegalHelper.kt:18]: Same here, extention of Context

**Review by kolyaopahle — APPROVED** [review]
lgtm 👍

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/formatter/FormatterInformationLegalHelper.kt (hunk 5 above)
Comment (BMItter) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/formatter/FormatterInformationLegalHelper.kt:18]: Not sure about this..
Methods of Formatter are meant to be used in layouts and extension funs can't be used there

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/formatter/FormatterHelper.kt (hunk 4 above)
Comment (BMItter) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/formatter/FormatterHelper.kt:20]: Not sure about this..
Methods of Formatter are meant to be used in layouts and extension funs can't be used there

**Review by axelherbstreith — APPROVED** [review]

**Pull request merged**